extractor.go

```diff
@@ -35,15 +35,11 @@ func AuthHeaderTokenExtractor(r *http.Request) (string, error) {
 func CookieTokenExtractor(cookieName string) TokenExtractor {
 	return func(r *http.Request) (string, error) {
 		cookie, err := r.Cookie(cookieName)
```

Comment:
From the docs:

```go
// Cookie returns the named cookie provided in the request or
// ErrNoCookie if not found.
// If multiple cookies match the given name, only one cookie will
// be returned.
func (r *Request) Cookie(name string) (*Cookie, error) {
	for _, c := range readCookies(r.Header, name) {
		return c, nil
	}
	return nil, ErrNoCookie
}
```

If there's no cookie, only ErrNoCookie will get returned, otherwise the *Cookie returned will never be nil but at worst `&http.Cookie{}`. This means we can greatly simplify the cookie token extractor logic.